test/e2e/framework/fixture.go

```diff
@@ -258,19 +263,84 @@ func NewWorkspaceWithWorkloads(t *testing.T, server RunningServer, orgClusterNam
 	return orgClusterName.Join(ws.Name)
 }
 
-// StartWorkspaceSyncer starts a new syncer to synchronize the identified set of
-// resource types scheduled for the given workload cluster from the upstream server
-// to the downstream server.
-func StartWorkspaceSyncer(
+// SyncerFixture contains the information to run a syncer fixture.
+type SyncerFixture struct {
+	RunningServer       RunningServer
+	KubeClient          kubernetes.Interface
+	upstreamConfig      *rest.Config
+	downstreamConfig    *rest.Config
+	resources           sets.String
+	logicalClusterName  logicalcluster.LogicalCluster
+	WorkloadClusterName string
+	wsClusterName       logicalcluster.LogicalCluster
+}
+
+// NewSyncerFixture creates a downstream server (fakeWorkloadServer), and then creates a workloadClusters on the provided upstream server
+// returns a SyncerFixture with the downstream server information, and its kubeclient.
+func NewSyncerFixture(
 	t *testing.T,
-	ctx context.Context,
 	resources sets.String,
-	workloadCluster *workloadv1alpha1.WorkloadCluster,
-	upstream, downstream RunningServer,
-) {
+	upstream RunningServer,
+	orgClusterName logicalcluster.LogicalCluster,
+	wsClusterName logicalcluster.LogicalCluster,
+) *SyncerFixture {
+	t.Log("Creating a fake workload server")
```

Comment:
(No action required) Maybe this log line should be in `NewFakeWorkloadServer` for consistency? I know `CreateWorkloadCluster` does that (that's why it didn't have a log line in any of the tests).